SCF-Assignment-5/Question3/NQueensProblem.java

```diff
@@ -0,0 +1,97 @@
+package Question3;
+
+/**
+ * The N Queen is the problem of placing N chess queens on an N�N chess board so
+ * that no two queens attack each other. (Thus, a solution requires that no two
+ * queens share the same row, column, or diagonal)
+ * 
+ * 
+ * @author Arjita
+ *
+ */
+public class NQueensProblem {
+	/**
+	 * 
+	 * check if queens can be placed in square matrix of given dimension when no
+	 * queen will attack each other.
+	 * 
+	 * @param board
+	 *            square matrix in which queens need to be placed
+	 * @param startRow
+	 *            row in which queen is to be placed
+	 * @param dimensionOfMatrix
+	 *            matrix dimension
+	 * @return true if queens can be placed when no queen attack each other
+	 *         otherwise false
+	 */
+	public static boolean nQueen(int[][] board, int startRow,
+			int dimensionOfMatrix) {
+		// check if all queens are placed
+		if (startRow == dimensionOfMatrix) {
+			// print the placed queen
+			for (int i = 0; i < dimensionOfMatrix; i++) {

```

Comment:
It is always advisable that we should never print the output in the function which is performing the calculations.
We should create a separate method for printing the result.